components/sections/Navbar/index.tsx

```diff
@@ -0,0 +1,93 @@
+import Image from 'next/image';
+import { useTheme } from 'next-themes';
+import type { FC } from 'react';
+import { useMemo } from 'react';
+
+import LanguageDropdown from '@/components/Common/LanguageDropDown';
+import ThemeToggle from '@/components/Common/ThemeToggle';
+import NavItem from '@/components/sections/NavItem';
+import { useLocale } from '@/hooks/useLocale';
+
+import style from './index.module.css';
+
+type NavItem = { key: string; link: string };
```

Comment:
The name for this type is used by a component imported here. Although, this does not cause any errors should we use a different name for the type or should we just merge the type where it's used.
If we have to rename, I could really use a suggestion here.

Reply:
This should be avoided. You should do:

```ts
type NavbarProps = { navItems: ComponentProps<NavItem>[] };
```

Reply:
The `NavItem` component accepts `children` as a prop which is displayed as text. 
so, I should do something like this.

```TS
type NavbarProps = {
  navItems: (Omit<ComponentProps<typeof NavItem>, 'children'> & {
    text: string;
  })[];
  localeConfig: ComponentProps<typeof LanguageDropdown>;
};
```